pkg/sriov/sriov.go

```diff
@@ -164,9 +164,11 @@ func (s *sriovManager) SetupVF(conf *sriovtypes.NetConf, podifName string, cid s
 		return "", fmt.Errorf("error setting temp IF name %s for %s", tempName, linkName)
 	}
 
+	macAddress := linkObj.Attrs().HardwareAddr.String()
 	// 3. Set MAC address
 	if conf.MAC != "" {
 		hwaddr, err := net.ParseMAC(conf.MAC)
+		macAddress = conf.MAC
```

Comment:
Can we move `macAddress` variable updates after the custom mac has applied successfully (after L181)?